GoogleForCreators/web-stories-wp
# Output: Preload the first page’s background media

## Context

LCP for stories could be improved through the right uses of preloading.

## Summary

Adds `link[preload]` elements in AMP HTML output.

Right now, this only includes the first page's background image/video/GIF and nothing else.

## Relevant Technical Choices



* Does not add `Link` HTTP headers as the benefit seems unnecessary and none of the other preloaded AMP scripts are added this way.

## To-do



Should more elements be included here, e.g. other large elements on the first page?

## User-facing changes



N/A

## Testing Instructions



1. Set an image/video/GIF as the first page's background
2. Preview story
3. Verify that a `link[preload]` element has been added for that image

### QA






This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #6742
